### ISSUE TITLE
TCA Hang on try_lock()

### ISSUE THREAD

**Issue by Mikefly123** [proveskit/circuitpy_flight_software#25]
# Context 
We use a TCA9548a multiplexer in the CubeSat as a means of dividing up our ```i2c1``` bus into multiple I2C busses for communicating with the sensors on each face of the satellite. With over 20 I2C devices in a fully built satellite, without the TCA we would be up a creek without a paddle in terms of efficiently fitting them all onto one bus. Not only would there be address collisions, the entire bus becomes extremely fragile as the failure of only one device could block every other device from using the bus. 

# Description of the Issue
What appears to be a [known but unsolved issue](https://github.com/adafruit/Adafruit_CircuitPython_TCA9548A/issues/29) with the CircuitPython TCA9548a library is that if something causes the try_lock() function to error out, an unhandled exception can cause the rest of your code to freeze until interrupted with a user input. 

Here is an example of what that error message may look like. When this happens the code completely freezes and does not advance. The code that causes this to happen generally looks like the following: 
```py 
try:
    tca.channel[0].try_lock()
    ...
except Exception as e:
    print(e)
    ...
```
<img width="1689" alt="Screenshot 2024-07-25 at 11 38 38 PM" src="https://github.com/user-attachments/assets/07676f4f-b27d-48f2-8489-77133b860096">

It appears that the unhandled ```OSError``` is what is causing the code to freeze up.

**Comment by Mikefly123** [comment]
My attempt to hot fix this is to specifically catch the OSError like so: 
```py
            try:
                self._scan_single_channel(channel, channel_to_face)
            except OSError:
                self.error_print("[ERROR][TCA] TCA try_lock failed. TCA may be malfunctioning.")
                self.hardware["TCA"] = False
                return
            except Exception as e:
                self.error_print(f"[ERROR][FACE]{traceback.format_exception(e)}")
```
It seems to work, but we should leave this issue open for more testing to verify.

**Comment by Mikefly123** [comment]
This does not seem to fix the issue, for whatever reason even if the TCA `try_lock` failure is caught by an exception it still hangs the entire software until a WDT reset: 
![image](https://github.com/user-attachments/assets/fad77c1a-f2da-4422-bbb8-f617c010e4ff)

**Comment by Mikefly123** [comment]
Actually it looks like catching the try_lock with the OS error does work! When the camera was added to PySquared though it added an edge case where we could hang on the try_lock in the camera init. 

#38 adds a try / except to fix this.

**Comment by Mikefly123** [comment]
I was rifling through the `tca9548a.py` and found this! 
```py
    def try_lock(self) -> bool:
        """Pass through for try_lock."""
        while not self.tca.i2c.try_lock():
            time.sleep(0)
        self.tca.i2c.writeto(self.tca.address, self.channel_switch)
        return True
```
I think this damn infinite loop is what is causing the Cube Killer bug!!!!

**Comment by Mikefly123** [comment]
This is possibly fixed in Commit [6f50227](https://github.com/proveskit/circuitpy_flight_software/commit/6f50227b2df932c927ad8d1d08456a3c0ccab3be)

**Comment by Mikefly123** [comment]
Closing this for now, as it does indeed appear to be fixed!
